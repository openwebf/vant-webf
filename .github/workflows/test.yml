name: CI

on:
  push:
    branches:
      - '**'

  pull_request:
    branches:
      - dev

  workflow_dispatch:

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run linter
      run: npm run lint

  test:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run test cases
      run: npm test

    - name: Upload coverage to Codecov  
      uses: codecov/codecov-action@v1
      with:
        token: ${{ secrets.CODECOV_TOKEN }}

  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Build
      run: npm run build
