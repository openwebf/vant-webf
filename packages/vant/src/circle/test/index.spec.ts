import { Circle } from '..';
import { mount, later } from '../../../test';

test('should update to final rate immediately if speed is 0', async () => {
  const wrapper = mount(Circle, {
    props: {
      rate: 50,
      currentRate: 0,
    },
  });

  await later();
  expect(wrapper.emitted('update:currentRate')).toBeTruthy();
});

test('should emit "update:currentRate" event during animation', async () => {
  const wrapper = mount(Circle, {
    props: {
      rate: 50,
      speed: 100,
    },
  });

  expect(wrapper.emitted('update:currentRate')).toBeFalsy();
  await later(50);
  expect(wrapper.emitted('update:currentRate')).toBeTruthy();
});

test('should change circle size when using size prop', () => {
  const wrapper = mount(Circle, {
    props: {
      size: 100,
    },
  });

  expect(wrapper.style.width).toEqual('100px');
  expect(wrapper.style.height).toEqual('100px');
});

test('should change stroke linecap when using stroke-linecap prop', () => {
  const wrapper = mount(Circle, {
    props: {
      strokeLinecap: 'square',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render start-position prop correctly', async () => {
  const wrapper = mount(Circle, {
    props: {
      startPosition: 'top',
    },
  });

  expect(wrapper.find('svg').style.transform).toEqual('');

  await wrapper.setProps({ startPosition: 'right' });
  expect(wrapper.find('svg').style.transform).toEqual('rotate(90deg)');

  await wrapper.setProps({ startPosition: 'bottom' });
  expect(wrapper.find('svg').style.transform).toEqual('rotate(180deg)');

  await wrapper.setProps({ startPosition: 'left' });
  expect(wrapper.find('svg').style.transform).toEqual('rotate(270deg)');
});
