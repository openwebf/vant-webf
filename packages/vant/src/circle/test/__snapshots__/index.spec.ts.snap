// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change stroke linecap when using stroke-linecap prop 1`] = `
<div class="van-circle">
  <svg viewbox="0 0 1040 1040">
    <path class="van-circle__layer"
          style="fill: none; stroke-width: 40px;"
          d="M 520 520 m 0, -500 a 500, 500 0 1, 1 0, 1000 a 500, 500 0 1, 1 0, -1000"
    >
    </path>
    <path d="M 520 520 m 0, -500 a 500, 500 0 1, 1 0, 1000 a 500, 500 0 1, 1 0, -1000"
          style="stroke-width: 41px; stroke-linecap: square; stroke-dasharray: 0px 3140px;"
          class="van-circle__hover"
    >
    </path>
  </svg>
</div>
`;
